src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,129 @@
+
+    private void run() {
+        ui.showGreeting();
+        Scanner sc = new Scanner(System.in);
+        scanner: while (sc.hasNextLine()) {
+            String[] parsedCommands;
+            Task task;
+            try {
+                parsedCommands = Parser.parse(sc.nextLine());
+            } catch (DukeException e) {
+                ui.showError(e);
+                continue;
+            }
+            try {
+                switch (parsedCommands[0]) {
+                case "list":
+                    if (parsedCommands.length == 1) {
+                        ui.showTasks(tasks);
+                    } else {
+                        ui.showTasks(tasks, parsedCommands[1]);
+                    }
+                    break;
+                
+                case "todo":
+                    tasks.add(new Todo(parsedCommands[1]));
+                    ui.showSuccess("Got it. I've added this todo:\n  %s\nNow you have %d tasks in the list.",
+                            tasks.get(tasks.size()), tasks.size());
+                    break;
+
+                case "deadline":
+                    tasks.add(new Deadline(parsedCommands[1], parsedCommands[2]));
+                    ui.showSuccess("Got it. I've added this deadline:\n  %s\nNow you have %d tasks in the list.",
+                            tasks.get(tasks.size()), tasks.size());
+                    break;
+
+                case "event":
+                    tasks.add(new Event(parsedCommands[1], parsedCommands[2]));
+                    ui.showSuccess("Got it. I've added this event:\n  %s\nNow you have %d tasks in the list.",
+                            tasks.get(tasks.size()), tasks.size());
+                    break;
+                
+                case "sort":
+                    tasks.sort();
+                    ui.showTasks(tasks);
+                    break;
+                
+                case "find":
+                    ui.showTasks(tasks.filter(parsedCommands[1]));
+                    break;
+
+                case "format":
+                    TimedTask.setFormat(parsedCommands[1]);
+                    ui.showTasks(tasks);
+                    break;
+
+                case "mark":
+                    task = tasks.mark(Integer.parseInt(parsedCommands[1]));
+                    ui.showSuccess("Nice! I've marked this task as done:\n  " + task);
+                    break;
+
+                case "unmark":
+                    task = tasks.unmark(Integer.parseInt(parsedCommands[1]));
+                    ui.showSuccess("OK, I've marked this task as not done yet:\n  " + task);
+                    break;
+                
+                case "delete":
+                    task = tasks.delete(Integer.parseInt(parsedCommands[1]));
+                    ui.showSuccess("Noted. I've removed this task:\n  %s\nNow you have %d tasks in the list.",
+                            task, tasks.size());
+                    break;
+    
+                case "bye":
```

Comment:
Consider adding a default branch to your switch statement as it is part of the code quality [guidelines](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-the-default-branch)